### PR TITLE
bt-7: Create Material UI theme with our custom typography settings

### LINKED ISSUE #7: Create Material UI theme with our custom typography settings
Create a Material UI Theme and modify our Typography component according to design

Figma: https://www.figma.com/file/Yd7USgP7PLvA0W794jHXWu/BSA_recruting?type=design&node-id=4120-2745&mode=design&t=mvb2tCBVBeWTg5yG-0
Info: https://mui.com/material-ui/customization/typography/

Image

### PR DESCRIPTION
**Pull request by nika-sirak**

### CODE REVIEW

**Comment by Simpsonkk** [comment]
I agree with the creation of a basic theme that will include different parts

**Review by Simpsonkk — APPROVED** [review]

File: frontend/src/bundles/common/theme/typography-theme.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography-theme.ts:1]: mb we should add size for global variables to use it in the app
`frontend/src/assets/css`

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/theme/typography.d.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        tub: React.CSSProperties;
+        input: React.CSSProperties;
+        label: React.CSSProperties;
+    }
+
+    interface TypographyVariantsOptions extends MuiTypographyVariantsOptions {
+        menu?: React.CSSProperties;
+        tub?: React.CSSProperties;
+        input?: React.CSSProperties;
+        label?: React.CSSProperties;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography.d.ts]: why are these fields not required?

File: frontend/src/bundles/common/theme/typography.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+}
+
+declare module '@mui/material/Typography' {
+    // eslint-disable-next-line unicorn/prevent-abbreviations
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography.d.ts]: Move to top of the file

File: frontend/src/bundles/common/theme/typography-theme.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        input: {
+            fontSize: 13,
+            lineHeight: 1.21,
+            fontWeight: 400,
+        },
+        label: {
+            fontSize: 13,
+            lineHeight: 1.21,
+            fontWeight: 500,
+        },
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography-theme.ts]: looks like we can add variables to use it in the scss

`assets/css/variables.scss`

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
In general looks good

File: frontend/src/bundles/common/theme/typography-theme.ts (hunk 3 above)
Comment (nika-sirak) [reply frontend/src/bundles/common/theme/typography-theme.ts]: Ok. I am working on it now.

Comment (nika-sirak) [reply frontend/src/bundles/common/theme/typography-theme.ts]: I created CSS variables and updated the typography part of the theme.

File: frontend/src/assets/css/variables.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    --font-size-h2: 28px;
+    --font-size-h3: 26px;
+    --font-size-h4: 24px;
+    --font-size-h5: 18px;
+    --font-size-h6: 16px;
+    --font-size-menu: 18px;
+    --font-size-body: 14px;
+    --font-size-btn: 16px;
+    --font-size-input: 13px;
+    --font-size-label: 13px;
```
Comment (nikita-remeslov) [inline frontend/src/assets/css/variables.scss]:
```suggestion
    --font-size-base: 13px;
```

File: frontend/src/bundles/common/theme/typography.d.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable unicorn/prevent-abbreviations */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography.d.ts]:
```suggestion
```

File: frontend/src/bundles/common/theme/typography-theme.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            fontWeight: 'var(--font-weight-medium)',
+        },
+        h4: {
+            fontSize: 'var(--font-size-h4)',
+            lineHeight: 'var( --line-height-h2-h4)',
+            fontWeight: 'var(--font-weight-medium)',
+        },
+        h5: {
+            fontSize: 'var(--font-size-h5)',
+            lineHeight: 'var( --line-height-h5-h6)',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography-theme.ts]:
```suggestion
            lineHeight: 'var(--line-height-h5-h6)',
```

File: frontend/src/bundles/common/theme/typography-theme.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            fontWeight: 'var(--font-weight-bold)',
+        },
+        h3: {
+            fontSize: 'var(--font-size-h3)',
+            lineHeight: 'var( --line-height-h2-h4)',
+            fontWeight: 'var(--font-weight-medium)',
+        },
+        h4: {
+            fontSize: 'var(--font-size-h4)',
+            lineHeight: 'var( --line-height-h2-h4)',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography-theme.ts]:
```suggestion
            lineHeight: 'var(--line-height-h2-h4)',
```

File: frontend/src/bundles/common/theme/typography-theme.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+            fontWeight: 'var(--font-weight-bold)',
+        },
+        h2: {
+            fontSize: 'var(--font-size-h2)',
+            lineHeight: 'var( --line-height-h2-h4)',
+            fontWeight: 'var(--font-weight-bold)',
+        },
+        h3: {
+            fontSize: 'var(--font-size-h3)',
+            lineHeight: 'var( --line-height-h2-h4)',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography-theme.ts]:
```suggestion
            lineHeight: 'var(--line-height-h2-h4)',
```

File: frontend/src/bundles/common/theme/typography-theme.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    typography: {
+        fontFamily: 'var(--font-family)',
+        h1: {
+            fontSize: 'var(--font-size-h1)',
+            lineHeight: 'var(--line-height-main)',
+            fontWeight: 'var(--font-weight-bold)',
+        },
+        h2: {
+            fontSize: 'var(--font-size-h2)',
+            lineHeight: 'var( --line-height-h2-h4)',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/theme/typography-theme.ts]:
```suggestion
            lineHeight: 'var(--line-height-h2-h4)',
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Comment by nikita-remeslov** [comment]
@nika-sirak Let's make something like this - https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/42/files
to follow the consistency

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**